### PR TITLE
NumberControl: Deprecate 36px default size

### PR DESCRIPTION
**Pull request by hbhalodia**
Part of #65751

## What?

Deprecate the 36px default size on NumberControl.

## Testing Instructions

- Unit tests pass
- Storybook stories should not log console warnings
- All code snippets in documentation (JSDoc, Storybook, README) should have the `__next40pxDefaultSize` prop enabled

### CODE REVIEW

**Comment by hbhalodia** [comment]
Hi @mirka, I have updated the PR with suggestions and feedbacks on the previous PRs, Can you please have a review for the same.

Also, I have looked in codebase regarding NumberControl usage in components, I have updated wherever I could find them. Let me know if there is any missing, I can update that as well.

Thank You 🙇.

File: packages/block-editor/src/components/line-height-control/index.js [hunk 1]
```diff
@@ -93,6 +93,7 @@ const LineHeightControl = ( {
 		<div className="block-editor-line-height-control">
 			<NumberControl
 				{ ...otherProps }
+				__shouldNotWarnDeprecated36pxSize
```
Comment (mirka) [inline packages/block-editor/src/components/line-height-control/index.js:96]: Nice ✨

File: packages/components/src/number-control/index.tsx [hunk 2]
```diff
@@ -34,6 +35,7 @@ function UnforwardedNumberControl(
 	forwardedRef: ForwardedRef< any >
 ) {
 	const {
+		__next40pxDefaultSize,
```
Comment (mirka) [inline packages/components/src/number-control/index.tsx]: This prop needs to be passed down to the `<Input>` to work (previously it was passed down as part of the `restProps`). For easy cleanup in the future, how about we do it this way?

```diff
diff --git a/packages/components/src/number-control/index.tsx b/packages/components/src/number-control/index.tsx
index b437e120b3..efa84879b8 100644
--- a/packages/components/src/number-control/index.tsx
+++ b/packages/components/src/number-control/index.tsx
@@ -35,7 +35,6 @@ function UnforwardedNumberControl(
 	forwardedRef: ForwardedRef< any >
 ) {
 	const {
-		__next40pxDefaultSize,
 		__unstableStateReducer: stateReducerProp,
 		className,
 		dragDirection = 'n',
@@ -62,7 +61,7 @@ function UnforwardedNumberControl(
 	maybeWarnDeprecated36pxSize( {
 		componentName: 'NumberControl',
 		size,
-		__next40pxDefaultSize,
+		__next40pxDefaultSize: restProps.__next40pxDefaultSize,
 		__shouldNotWarnDeprecated36pxSize,
 	} );
 
```

File: packages/components/src/angle-picker-control/index.tsx [hunk 3]
```diff
@@ -57,6 +57,7 @@ function UnforwardedAnglePickerControl(
 		<Flex { ...restProps } ref={ ref } className={ classes } gap={ 2 }>
 			<FlexBlock>
 				<NumberControl
+					__next40pxDefaultSize
```
Comment (mirka) [inline packages/components/src/angle-picker-control/index.tsx:60]: FYI, either having the `size="__unstable-large"` or `__next40pxDefaultSize` is good enough, so we can remove one of them. (`size="__unstable-large"` was an older way to opt into this size 😄)

File: packages/components/src/color-picker/input-with-slider.tsx [hunk 4]
```diff
@@ -31,6 +31,7 @@ export const InputWithSlider = ( {
 	return (
 		<HStack spacing={ 4 }>
 			<NumberControlWrapper
+				__next40pxDefaultSize
```
Comment (mirka) [inline packages/components/src/color-picker/input-with-slider.tsx:34]: This one also has an `size="__unstable-large"`.

**Review by mirka — COMMENTED** [review]
Great work, thank you for applying all the previous tips!

File: packages/components/src/number-control/index.tsx (hunk 2 above)
Comment (hbhalodia) [reply packages/components/src/number-control/index.tsx]: This is now resolved.

File: packages/components/src/angle-picker-control/index.tsx (hunk 3 above)
Comment (hbhalodia) [reply packages/components/src/angle-picker-control/index.tsx:60]: This is now addressed, removed `size="__unstable-large"` and used the new way to add the size.

File: packages/components/src/color-picker/input-with-slider.tsx (hunk 4 above)
Comment (hbhalodia) [reply packages/components/src/color-picker/input-with-slider.tsx:34]: This is now addressed, removed `size="__unstable-large"` and used the new way to add the size.

**Comment by hbhalodia** [comment]
Hi @mirka, All the issues have been addressed.

Thank You,

**Review by mirka — APPROVED** [review]
Perfect, thanks!

**Pull request merged**